docs/framework/angular/angular-table.md

```diff
@@ -42,23 +50,168 @@ Example:
               let cell
             "
-          >
-            <div [innerHTML]="cell"></div>
-          </ng-container>
-        </td>
-      }
-    </tr>
-  }
+  >
+    <!-- if you want to render a simple string -->
+    {{ cell }}
+    <!-- if you want to render an html string -->
+    <div [innerHTML]="cell"></div>
+  </ng-container>
+</td>
+}
+</tr>
+}
 </tbody>
 ```
+
+#### Rendering a TemplateRef
+
+In order to render a TemplateRef into a specific column header/cell/footer, you can pass the TemplateRef into the column
+definition.
+
+You can access the TemplateRef data via the `$implicit` property, which is valued based on what is passed in the props
+field of flexRender.
+
+In most cases, each TemplateRef will be rendered with the $implicit context valued based on the cell type in this way:
+
+- Header: `HeaderContext<T, ?>`
+- Cell: `CellContext<T, ?>`,
+- Footer: `HeaderContext<T, ?>`
+
+```angular17html
+
+<ng-container
+  *flexRender="
+              cell.column.columnDef.cell;
+              props: cell.getContext();
+              let cell
+            "
+>
+  <!-- if you want to render a simple string -->
+  {{ cell }}
+  <!-- if you want to render an html string -->
+  <div [innerHTML]="cell"></div>
+</ng-container>
+
+<ng-template #myCell let-context>
+  <!-- render something with context -->
+</ng-template>
+```
+
+Full example:
+
+```ts
+import type {
+  CellContext,
+  ColumnDef,
+  HeaderContext,
+} from '@tanstack/angular-table'
+import {Component, TemplateRef, viewChild} from '@angular/core'
+
+@Component({
+  template: `
+    <tbody>
+      @for (row of table.getRowModel().rows; track row.id) {
+        <tr>
+          @for (cell of row.getVisibleCells(); track cell.id) {
+            <td>
+              <ng-container
+                *flexRender="
+                  cell.column.columnDef.cell;
+                  props: cell.getContext(); // Data given to the TemplateRef
+                  let cell
+                "
+              >
+                <!-- if you want to render a simple string -->
+                {{ cell }}
+                <!-- if you want to render an html string -->
+                <div [innerHTML]="cell"></div>
+              </ng-container>
+            </td>
+          }
+        </tr>
+      }
+    </tbody>
+
+    <ng-template #customHeader let-context>
+      {{ context.getValue() }}
+    </ng-template>
+    <ng-template #customCell let-context>
+      {{ context.getValue() }}
+    </ng-template>
+  `,
+})
+class AppComponent {
+  customHeader =
+    viewChild.required<TemplateRef<{ $implicit: HeaderContext<any, any> }>>(
+      'customHeader'
+    )
+  customCell =
+    viewChild.required<TemplateRef<{ $implicit: CellContext<any, any> }>>(
+      'customCell'
+    )
+
+  columns: ColumnDef<unknown>[] = [
+    {
+      id: 'customCell',
+      header: () => this.customHeader(),
+      cell: () => this.customCell(),
+    },
+  ]
+}
+```
+
+#### Rendering a TemplateRef
```

Comment:
why does this have the same title as the heading above?

Reply:
wrong copy paste, fixed it